stringer.go

```diff
@@ -66,3 +54,34 @@ func (k key) String() string {
 	}
 	return k.t.String()
 }
+
+func (pl paramList) String() string {
+	args := make([]string, len(pl.Params))
+	for i, p := range pl.Params {
+		args[i] = p.String()
+	}
+	return fmt.Sprint(args)
+}
+
+func (sp paramSingle) String() string {
+	// ~tally.Scope means optional
+	// ~tally.Scope:foo means named optional
+
+	var prefix, suffix string
+	if sp.Optional {
+		prefix = "~"
+	}
+	if sp.Name != "" {
+		suffix = ":" + sp.Name
+	}
+
+	return fmt.Sprintf("%s%v%s", prefix, sp.Type, suffix)
+}
+
+func (op paramObject) String() string {
+	fields := make([]string, len(op.Fields))
+	for i, f := range op.Fields {
+		fields[i] = f.Param.String()
+	}
+	return strings.Join(fields, " ")
```

Comment:
![so-what](https://user-images.githubusercontent.com/972790/31161206-6cc48406-a88a-11e7-8d0e-f1c546b69465.gif)

What, we suddenly care about the overhead of `fmt.Sprint`? Srsly tho let's use one or the other.

Reply:
Discussed offline. We're doing this to preserve current behavior exactly as-is. We can iterate on this in the future.